packages/material-ui/src/Typography/Typography.js

```diff
@@ -35,6 +36,8 @@ export const styles = (theme) => ({
   subtitle2: theme.typography.subtitle2,
   /* Styles applied to the root element if `variant="overline"`. */
   overline: theme.typography.overline,
+  /* Styles applied to the root element if `variant="inherit"`. */
+  inherit: {},
```

Comment:
For consistency I've added this variant `classKey`. This way we don't need to update the warnings as the variant classKey will always exists. If we don't want to make this change, I can work on the warning.